cloudera/datadog_checks/cloudera/api_client_factory.py

```diff
@@ -0,0 +1,34 @@
+import cm_client
+import cm_client.rest
+import packaging.version
+
+from datadog_checks.base import ConfigurationError
+from datadog_checks.cloudera.api_client_v7 import ApiClientV7
+
+
+def make_api_client(check, config):
+    cm_client.configuration.username = config.workload_username
+    cm_client.configuration.password = config.workload_password
+    api_client = cm_client.ApiClient(config.api_url)
```

Comment:
I noticed that there aren't any validations for whether the api_url has the api version appended. What happens when the `cm_client.ApiClient` is initialized with a bad url? Can we add exception handling for that?

Reply:
Oh I see, we end up throwing the ConfigurationError [here](https://github.com/DataDog/integrations-core/pull/13244/files#diff-58bc7b8b4f1e633275261c9b9b292716bd2290a7ddc1b216b411af04383bfee5R34)?

Reply:
If a bad `api_url` is used (either a misspelled URL or a URL without the api version at the end), the check will raise an exception and provide the following service check:
```
[
  {
    "check": "cloudera.can_connect",
    "host_name": "COMP-C02F513WML87",
    "timestamp": 1671480853,
    "status": 2,
    "message": "Cloudera API Client is none: (404)\nReason: Not Found\nHTTP response headers: HTTPHeaderDict({'Server': 'nginx', 'Date': 'Mon, 19 Dec 2022 20:14:13 GMT', 'Content-Length': '0', 'Connection': 'keep-alive', 'x-response-nginx': 'true'})\n",
    "tags": [
      "api_url=https://spark-test-gateway.agent-in.jfha-h5rc.a0.cloudera.site"
    ]
  }
]
```

Unfortunately Cloudera doesn't return a specific error for a valid API URL without the API version in the background, so there wouldn't be an easy way to distinguish between that and just an incorrect URL. 

I think one possibility could be the following:
- Attempt a connection with `api_url` value
- if it failed, try again with a default `/v48` appended at the end.

Reply:
> Adding a validation for the url would be nice too with logging to tell users to append the api version to the URL would be helpful

I think that would be a good idea, although it may be a bit hack-y trying to read the last section of the URL to check if it is in the format of `v<API_VERSION>`. Cloudera also periodically updates the version number, so we wouldn't be able to check only for `v48` at the end.

> Oh I see, we end up throwing the ConfigurationError [here](https://github.com/DataDog/integrations-core/pull/13244/files#diff-58bc7b8b4f1e633275261c9b9b292716bd2290a7ddc1b216b411af04383bfee5R34)?
That line is more for raising an error after the agent was able to connect to the API but the Cloudera Manager version is unsupported, i.e. If we successfully connect to Cloudera but Cloudera Manager is v5 instead of v7.

Reply:
That makes sense! Maybe we can keep the current implementation and then add debug logging suggesting to double-check that the api_url has the version appended.

Reply:
Fixed in https://github.com/DataDog/integrations-core/pull/13244/commits/30dfbbf87a26e1dc16f4eafde0e3a74e67da2e6c